libsweep/src/sweep.cc

```diff
@@ -49,54 +51,49 @@ void sweep_error_destruct(sweep_error_s error) {
   delete error;
 }
 
-sweep_device_s sweep_device_construct_simple(const char* port, sweep_error_s* error) {
+sweep_device_s sweep_device_construct_simple(const char* port, sweep_error_s* error) try {
   SWEEP_ASSERT(error);
-
   return sweep_device_construct(port, 115200, error);
+} catch (const std::exception& e) {
+  *error = sweep_error_construct(e.what());
```

Comment:
(Un) Luckily I ran into the same problem during my initial attempt to introduce exceptions